### DISCUSSION TITLE
Can a DELETE hook be called also via a relational trigger?

### DISCUSSION THREAD

**Discussion by AlexVanK** [directus/directus#13392] [Archive]
Hi,

This is my custom HOOK which is fired correctly whenever I delete an item in the _administrative_attachments_ collection:
![image](https://user-images.githubusercontent.com/56647386/169121329-84c03a8b-0859-4585-b3b0-4dfbf6aad809.png)

However the _administrative_attachment_ collection is a connected with the parent _application_info_ collection via a M2O relationship (or O2M if looked from the perspective of the parent collection).
These are the current relationship triggers:
![image](https://user-images.githubusercontent.com/56647386/169123586-36f51dc3-da8d-4abf-a86a-68173939dcb9.png)

So if I delete an _application_info_ item, the relationship trigger deletes also the _administrative_attachment_ item, but this action doesn't fire the hook defined above...

How to go about this use case?

**Comment by rijkvanzanten** [comment]
The cascade delete is currently handled through the database foreign key constraint that deletes the item, so it's actually not Directus that deletes that item (and hence doesn't fire the hook) 🤔

**Reply by AlexVanK** [reply]
Understood... 
Do you have a suggestion on how to go about this particular use case?

In other words what I am trying to do is delete a file that is related to a related collection:

_Collection_A_ O2M _Collection_B_ O2M _Directus_files_

upon deleting A, I want to delete B and that works...
but I also want to delete the associated file connected through _Collection_B_

**Reply by rijkvanzanten** [reply]
I think the next best approach would be to setup a custom hook that listens for that delete, and then handle the file deletion yourself 🤔

**Reply by AlexVanK** [reply]
Hum... this sounds a bit convoluted...

1) Get the current collection upon deletion through the .delete hook
2) Then get the IDs of the related collections
2b) What if I had another level deep?
3) Then get the IDs of the files related to the related collections
4) Delete the files from disk...

So this is the code (for now)...

![image](https://user-images.githubusercontent.com/56647386/172220095-be023861-f03f-40fd-b6d3-787cd16c9a7b.png)

However it seems an odd procedure to loop over child collections in order to dig out the file ids and then delete them manually...

Let me know if there is a better way

**Reply by NilsBaumgartner1994** [reply]
I am facing the same challenge now. I created a doc enchancement issue: https://github.com/directus/directus/issues/24662
In https://github.com/directus/directus/issues/2942 was a workaround/hack described, to set the id to NULL and then check in a loop for items with relational_id NULL and then to delete the file.
